### ISSUE TITLE
multi-platform release

### ISSUE THREAD

**Issue by headupinclouds** [elucideye/drishti#291]
see: `./bin/build-all-parallel.sh`

- [x] ~~~bundling dependencies requires hunter package list feature~~~
- [x] polly log.txt conflicts on parallel builds (use `_logs/<toolchain>/log*.txt`)
- [ ] `--pack FORMAT` alters the installation behavior when run with `--strip` or `--install`
- [x] confirm unique installation paths support parallel device testing (iOS and Android)
- [x] see `./bin/license.sh` for wget license patching (TODO: add hunter license issue link)

**Comment by ruslo** [comment]
> polly log.txt conflicts on parallel builds (use _logs/<toolchain>/log*.txt)

Fixed as far as I know

**Comment by ruslo** [comment]
> see ./bin/license.sh for wget license patching (TODO: add hunter license issue link)

Hunter part resolved here, do we need to change anything in Drishti?

**Comment by headupinclouds** [comment]
That was a temporary workaround.  I think it may be redundant with the path extensions you added to the hunter license feature.

**Comment by headupinclouds** [comment]
There is probably significant overlap between this issue and the "Thaw" (python?) feature.

**Comment by headupinclouds** [comment]
> confirm unique installation paths support parallel device testing

This was mostly about confirming that parallel builds with a shared/tethered mobile test device would be ok (i.e., two different Android toolchain builds + tests on the same device).  I wasn't sure about this, but I noticed the the toolchain suffix is currently added to the installation path for Android and the BUNDLE_ID for iOS apps, so I think this one is good.

**Comment by headupinclouds** [comment]
> bundling dependencies requires hunter package list feature

Exporting dependencies is addressed to some extent by the merged static lib.  As noted in hunter discussions on this topic, it is best to just use hunter if possible.  I'll remove this one from current priorities.

**Comment by headupinclouds** [comment]
> --pack FORMAT alters the installation behavior when run with --strip or --install

I don't recall exact details.  I guess this would become apparent if `--pack` is tested in the current multi-platform build scripts.

**Comment by ruslo** [comment]
> There is probably significant overlap between this issue and the "Thaw" (python?) feature.

I see the Thaw as a tool to substitute all current `*.sh` scripts. Currently listed issues are not directly related to this.

**Comment by ruslo** [comment]
>  I guess this would become apparent if --pack is tested in the current multi-platform build scripts.

`--pack` run CPack which currently doesn't support iOS:
* https://gitlab.kitware.com/cmake/cmake/issues/16149

We have to use `polly.py --archive` anyway.

**Comment by headupinclouds** [comment]
Agreed.  I guess this issue's title could have been`multi-platform release quirks`, and a separate issue like `multi-platform release improvement/refactoring/design (?)` could be added or included as part of the "Thaw" feature.  

This was a "catch-all" list of bugs/quirks I noticed during a multi-platform build.  I think most of these are answered and/or fixed, apart from the `--pack` issue (unfortunately I didn't include many details when I posted it).

I think it makes sense to close this one and address the long term multi-platform build requirements/design in a separate top-down issue (requirements, etc).
